src/MPoly.jl

```diff
@@ -1271,69 +1297,57 @@ end
 ###############################################################################
 
 @doc Markdown.doc"""
-    polynomial_ring(R::AbstractAlgebra.Ring, s::Vector{T}; cached::Bool = true, ordering::Symbol = :lex) where T <: Union{String, Char, Symbol}
-
-Given a base ring `R` and an array of strings `s` specifying how the
-generators (variables) should be printed, return a tuple `T, (x1, x2, ...)`
-representing the new polynomial ring $T = R[x1, x2, ...]$ and the generators
-$x1, x2, ...$ of the polynomial ring. By default the parent object `T` will
-depend only on `R` and `x1, x2, ...` and will be cached. Setting the optional
-argument `cached` to `false` will prevent the parent object `T` from being
-cached. `S` is a symbol corresponding to the ordering of the polynomial and
-can be one of `:lex`, `:deglex` or `:degrevlex`.
+    polynomial_ring(R::Ring, s::Vector{T}; cached::Bool = true, ordering::Symbol = :lex) where T <: Union{Symbol, AbstractString, Char}
```

Comment:
Another meta comment (for a future PR, not this one): there a bunch of places that use `Union{Symbol, AbstractString, Char}` for this very purpose; even more exist that use this, or *should* use this (but instead hardcode `Symbol` or `String`, or some other subset) in Nemo, Oscar, perhaps also Hecke.

So perhaps we should introduce a name for this. Say `const VarName = Union{Symbol, AbstractString, Char}` or so?

And then going one beyond, there are a bunch of places that which to specify a vector of variable names. Some just allow for a simple `Vector{VarName}` (or often only `Vector{String}` or `Vector{Symbol}`. But we also have code that allows sophisticated things allowing stuff like `R, x, y = polynomial_ring(QQ, :x => 1:3, :y => 1:2)`. Anyway, some code supports this, but other does not. Perhaps we could factor out the logic for this so that it can be used by `polynomial_ring`, but also by other such functions (universal polynomial rings, Laurent polynomials, free associative algebras, free groups, etc.).

Again: not for this PR! Just to get you primed for future work ;-)